### ISSUE TITLE
[Feature request] Serving custom html only for 404s

### ISSUE THREAD

**Issue by Fukitsu** [svenstaro/miniserve#753]
Right now if you specify a custom index file it will serve that for all paths. I'd like to be able to serve a custom index.html only for when an error occurs (404 in this case)

**Comment by aliemjay** [comment]
Have you tried `--spa` option?

**Comment by Fukitsu** [comment]
Yes, but that servers the index.html for all routes instead of the default listing. I'd like to serve only a custom html only for when a 404 occurrs
